save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/demo/CpgView.kt

```diff
@@ -136,3 +177,82 @@ val cpgView: VFC = VFC {
         }
     }
 }
+
+@Suppress("TOO_LONG_FUNCTION", "LongMethod")
+private fun ChildrenBuilder.displayCpgNodeAdditionalInfo(
+    nodeName: String?,
+    applicationName: String,
+    additionalInfo: CpgNodeAdditionalInfo?,
+    setSelectedNodeName: (String?) -> Unit,
+) {
+    div {
+        className = ClassName("card card-body p-0")
+        table {
+            thead {
+                tr {
+                    className = ClassName("bg-dark text-light")
+                    th {
+                        scope = "col"
+                        +"Name"
+                    }
+                    th {
+                        scope = "col"
+                        +(nodeName ?: NOT_PROVIDED).formatPathToFile(applicationName)
+                    }
+                    button {
+                        className = ClassName("btn p-0 position-absolute")
+                        fontAwesomeIcon(faTimesCircle)
+                        type = "button".unsafeCast<ButtonType>()
+                        onClick = { setSelectedNodeName(null) }
+                        style = jso {
+                            top = "1%".unsafeCast<Top>()
+                            right = "1%".unsafeCast<Right>()
+                        }
+                    }
+                }
+            }
+            tbody {
+                listOf(
```

Comment:
Maybe make this a map of columns names to accessor functions and extract into a file-level variable? Then each `value` of the map can consume `additionalInfo` and return its result, and the map will be reused between rerenders.

Reply:
Ok, will do it

Reply:
Done